addons/stock/models/stock_picking.py

```diff
@@ -1155,8 +1155,11 @@ def _action_done(self):
 
     def _send_confirmation_email(self):
         subtype_id = self.env['ir.model.data']._xmlid_to_res_id('mail.mt_comment')
-        for stock_pick in self.filtered(lambda p: p.company_id.stock_move_email_validation and p.picking_type_id.code == 'outgoing'):
+        for stock_pick in self.filtered(lambda p: p.company_id.stock_move_email_validation and p.picking_type_id.code in ('outgoing', 'dropship')):
```

Comment:
If you need to do specific things for dropships, then do it in `stock_dropshipping`.